CNV-21825: latency checkup permissions and API update (#53437)
openshift/openshift-docs: modules/virt-measuring-latency-vm-secondary-network.adoc

[id="virt-measuring-latency-vm-secondary-network_{context}"]
= Checking network connectivity and latency for virtual machines on a secondary network

> You use a predefined checkup to verify network connectivity and measure latency between virtual machines (VMs) that are attached to a secondary network interface.

Reviewer: ```suggestion
You use a predefined checkup to verify network connectivity and measure latency between two virtual machines (VMs) that are attached to a secondary network interface.
```

kind: ConfigMap
metadata:
  name: kubevirt-vm-latency-checkup-config
data:
  spec.timeout: 5m

>   spec.serviceAccountName: vm-latency-checkup-sa

Reviewer: The `serviceAccountName` is specified on the checkup's Job object.
```suggestion
```

>   spec.param.network_attachment_definition_namespace: "default"

Reviewer: The NetworkAttachmentDefinition is expected to be at the same namespace as the checkup.
```suggestion
  spec.param.network_attachment_definition_namespace: <target_namespace>
```

  spec.param.network_attachment_definition_name: "blue-network" <1>
  spec.param.max_desired_latency_milliseconds: "10" <2>
  spec.param.sample_duration_seconds: "5" <3>
  spec.param.source_node: "worker1" <4>
  spec.param.target_node: "worker2" <5>
----
<1> The name of the `NetworkAttachmentDefinition` object.
<2> Optional: The maximum desired latency, in milliseconds, between the virtual machines. If the measured latency exceeds this value, the checkup fails.
<3> Optional: The duration of the latency check, in seconds.
<4> Optional: When specified, latency is measured from this node to the target node. If the source node is specified, the `spec.param.target_node` field cannot be empty.
<5> Optional: When specified, latency is measured from the source node to this node.

> . Apply the config map manifest in the framework’s namespace:

Reviewer: ```suggestion
. Apply the config map manifest in the target namespace:
```

kind: Job
metadata:
  name: kubevirt-vm-latency-checkup
spec:
  backoffLimit: 0
  template:
    spec:
      serviceAccountName: vm-latency-checkup-sa
      restartPolicy: Never
      containers:
        - name: vm-latency-checkup

>           image: registry.redhat.io/container-native-virtualization/checkup-framework:v4.12.0

Reviewer: ```suggestion
          image: registry.redhat.io/container-native-virtualization/vm-network-latency-checkup:v4.12.0
```

          securityContext:
            allowPrivilegeEscalation: false
            capabilities:
              drop: ["ALL"]
            runAsNonRoot: true
            seccompProfile:
              type: "RuntimeDefault"
          env:
            - name: CONFIGMAP_NAMESPACE
              value: <target_namespace>
            - name: CONFIGMAP_NAME
              value: kubevirt-vm-latency-checkup-config

>             - name: POD_UID
>               valueFrom:
>                 fieldRef:
>                   fieldPath: metadata.uid

Reviewer: The `POD_UID` environment variable is not supported in 4.12.
It will be relevant in the next version.
```suggestion
```

  name: kubevirt-vm-latency-checkup-config
  namespace: <target_namespace>
data:
  spec.timeout: 5m

>   spec.param.network_attachment_definition_namespace: "default"

Reviewer: ```suggestion
  spec.param.network_attachment_definition_namespace: <target_namespace>
```
